cogs/audio.py

```diff
@@ -627,13 +629,21 @@ def _is_queue_playlist(self, server):
 
     async def _join_voice_channel(self, channel):
         server = channel.server
+        connect_time = self.connect_timers.get(server.id, 0)
+        if time.time() < connect_time:
+            diff = int(connect_time - time.time())
+            raise ConnectTimeout("You are on connect cooldown for another {}"
+                                 " seconds.".format(diff))
         if server.id in self.queue:
             self.queue[server.id]["VOICE_CHANNEL_ID"] = channel.id
         try:
-            await self.bot.join_voice_channel(channel)
+            await asyncio.wait_for(self.bot.join_voice_channel(channel),
```

Comment:
This line is guaranteed to throw `asyncio.futures.TimeoutError` if it does in fact time out.